freqtrade/edge/__init__.py

```diff
@@ -52,8 +53,17 @@ def __init__(self, config: Dict[str, Any], exchange, strategy) -> None:
 
         self.edge_config = self.config.get('edge', {})
         self._cached_pairs: Dict[str, Any] = {}  # Keeps a list of pairs
+        self._final_pairs: list = []
 
-        self._total_capital: float = self.config['stake_amount']
+        # checking max_open_trades. it should be -1 as with Edge
+        # the number of trades is determined by position size
+        if self.config['max_open_trades'] != -1:
+            logger.critical('max_open_trades should be -1 in config !')
+
+        if self.config['stake_amount'] != constants.UNLIMITED_STAKE_AMOUNT:
+            raise OperationalException('Edge works only with unlimited stake amount')
```

Comment:
do we need to force this onto users?
i think edge would benefit users also if they wanna work with fixed stakes, which this PR makes impossible.

Yes, some profit may be left on the table - but that's the users decision, shouldn't be the bot's i think.

Reply:
This is only as a hint to user. so that he knows the amount is not fixed and will be adjusted by Edge. I thought about how I can pass the control to user on the amount, I didn't find any reasonable solution other than capital_percentage. 

Using a fixed stake_amount is dangerous and may break Edge basic functionality. 
Your initial stake_amount is 10 ETH but then after two trades you lose two trades ... your position size is calculated again based on 10ETH which is dangerous and does not reflect basics of position sizing. 

The reverse is true as well. if after two trades your available capital is 12ETH then your position size should reflect it. 

The only reasonable way to limit the stake amount is to fix the percentage of your capital that is allowed for trading (which is included in config). 

Happy to hear your counter argument though if any.

Reply:
nah it's fine, you're right ... :+1: 

the only thing - MAYBE (not in this PR) we could add in a max position size of 10ETH - so in your example, after i won - the 2 ETH i won are not touched by the bot

but it's nothing i'm insisting on ... just an idea ...

Reply:
It breaks the logic ... position size is the goal. limiting it, is misleading. I am not in favour tbh.